src/ext/java/org/opentripplanner/ext/flex/FlexRouter.java

```diff
@@ -95,21 +96,32 @@ public FlexRouter(
 
     Set<StopLocation> egressStops = streetEgressByStop.keySet();
 
-    Collection<Itinerary> itineraries = new ArrayList<>();
+    Map<FlexTrip, Itinerary> itinerariesByTrip = new HashMap<>();
 
     for (FlexAccessTemplate template : this.flexAccessTemplates) {
       StopLocation transferStop = template.getTransferStop();
+      FlexTrip trip = template.getFlexTrip();
       if (egressStops.contains(transferStop)) {
-        for(NearbyStop egress : streetEgressByStop.get(transferStop)) {
-          Itinerary itinerary = template.createDirectItinerary(egress, arriveBy, departureTime, startOfTime);
-          if (itinerary != null) {
-            itineraries.add(itinerary);
+        for (NearbyStop egress : streetEgressByStop.get(transferStop)) {
+          Itinerary itinerary = template.createDirectItinerary(
```

Comment:
Check how many itineraries are initially created here, and if we should prevent that from happening earlier.

Reply:
This has been reverted, and will probably be part of the filtering in a future PR.